pkg/pillar/types/ifnametopci.go

```diff
@@ -79,6 +79,16 @@ func PCISameController(long1 string, long2 string) bool {
 	return ctrl1 == ctrl2
 }
 
+// PCIGetIOMMU returns IOMMU group tag as seen by the control domain
+func PCIGetIOMMU(long string) (string, error) {
```

Comment:
is `long` here a host bdf, or guest bdf? So far I thought we are operating guest addresses. But now I am confused.

Reply:
This is for checks done on the host side (from the domainmgr microservice).

Reply:
That is actually clear. Let me try to explain what confuses me . 

The host eventually sets up the passthrough of a device identified by "bus:device:function", and guest will get this device under its own `bdf`.

As in the previous patch:
```
[device]
   driver = "vfio-pci"
   host = "{{.PciShortAddr}}"
   bus = "pci.{{.PCIId}}"
   addr = "0x0"
```
does `long` here correspond to `host`, or `bus` + `addr`?

Do I understand correctly, that check `PCISameController` does not allow devices sitting on the same bus be allocated to one guest? Which would work if this limit applies to the guest buses (I do not know why it is necessary though). But many machines do have just one physical PCI bus (e.g. Raspberry), so if this limitation applies to the host address, then we shall be able to run only one guest with a pass through PCI.

Now, if `long` describes a guest device, then function `PCIGetIOMMU` might be not correct.

Reply:
Long here corresponds to the fully qualified PCI address starting with domain on the host. PCISameController attempts to guess (and does a pretty good job with iommu groups when that is available)

Reply:
> Do I understand correctly, that check PCISameController does not allow devices sitting on the same bus be allocated to one guest?

It is more subtle -- we need to make sure that assigning device A wouldn't render device B inoperable. A simple comparison of a PCI address doesn't really work (hence this patch). What works is a combination of iommu group information and, sadly, at times bugs in hardware @zededa-yuri 

> But many machines do have just one physical PCI bus (e.g. Raspberry), so if this limitation applies to the host address, then we shall be able to run only one guest with a pass through PCI.

Again -- as long as iommu groups (well there are not iommu groups on ARM -- but still) allow such an assignment we should be fine. Which is to say -- we're mostly fine on x86, as for ARM -- somebody will need to dig much more into how to make the above statement "we need to make sure that assigning device A wouldn't render device B inoperable" effectively checkable there. That would be awesome, but not part of this PR.